External CI: Boilerplate code for aggregate build pipeline

jayhawk-commits opened the pull request:
- Add knobs to toggle aggregate build options.
- Aggregate build pipeline will pull ROCm dependencies from earlier in the same pipeline.
- Changing build pool of some components for more compute power.
- Deleting deprecated component.

danielsu-amd commented on .azuredevops/components/AMDMIGraphX.yml:
```diff
@@ -132,12 +138,20 @@ jobs:
+      succeeded(),
+      eq(variables.ENABLE_GFX942_TESTS, 'true'),
+      not(containsValue(split(variables.DISABLED_GFX942_TESTS, ','), variables['Build.DefinitionName'])),
+      eq(variables['aggregatePipeline'], false)
```
I think `${{ eq(parameters.aggregatePipeline, false) }}` would work here, like in this condition https://github.com/ROCm/ROCm/blob/develop/.azuredevops/templates/steps/docker-container.yml#L110

danielsu-amd committed:
External CI: Boilerplate code for aggregate build pipeline
- Add knobs to toggle aggregate build options.
- Aggregate build pipeline will pull ROCm dependencies from earlier in the same pipeline.
- Changing build pool of some components for more compute power.
- Deleting deprecated component.
- Add Ninja to dependency compilation in MIOpen.
- Add retries to wget for MIOpen CK build case.

danielsu-amd reviewed (dismissed):
Had to rebase basically everything, I saved the old branch state here for now: https://github.com/ROCm/ROCm/tree/amd/danielsu/jmacaran/aggregatePipeline

darren-amd commented on .azuredevops/components/amdsmi.yml:
```diff
@@ -59,6 +64,7 @@ jobs:
       and(succeeded(),
         eq(variables['ENABLE_${{ upper(job.target) }}_TESTS'], 'true'),
         not(containsValue(split(variables['DISABLED_${{ upper(job.target) }}_TESTS'], ','), variables['Build.DefinitionName']))
```
Missing a comma?

danielsu-amd replied:
Added it in

darren-amd commented on .azuredevops/components/hipBLASLt.yml:
```diff
@@ -98,13 +103,7 @@ jobs:
         checkoutRef: ${{ parameters.checkoutRef }}
         dependencyList: ${{ parameters.rocmDependencies }}
         gpuTarget: ${{ job.target }}
-    - task: Bash@3
```
Was this mean to be removed?

danielsu-amd replied:
Added it back in

danielsu-amd committed:
Address comments

darren-amd reviewed (approved):
LGTM
